async-openai/src/embedding.rs

```diff
@@ -23,14 +27,56 @@ impl<'c, C: Config> Embeddings<'c, C> {
         &self,
         request: CreateEmbeddingRequest,
     ) -> Result<CreateEmbeddingResponse, OpenAIError> {
-        self.client.post("/embeddings", request).await
+        if !matches!(request.encoding_format, Some(EncodingFormat::Base64)) {
+            self.client.post("/embeddings", request).await
+        } else {
+            let response: CreateEmbeddingResponse<Base64Embedding> =
+                self.client.post("/embeddings", request).await?;
+            response.try_into()
+        }
+    }
+}
+
+impl TryFrom<CreateEmbeddingResponse<Base64Embedding>> for CreateEmbeddingResponse<Embedding> {
+    type Error = OpenAIError;
+
+    fn try_from(response: CreateEmbeddingResponse<Base64Embedding>) -> Result<Self, Self::Error> {
+        let response = CreateEmbeddingResponse {
+            model: response.model,
+            object: response.object,
+            usage: response.usage,
+            data: response
+                .data
+                .into_iter()
+                .map(Embedding::try_from)
+                .collect::<Result<_, _>>()?,
+        };
+        Ok(response)
+    }
+}
+
+impl TryFrom<Base64Embedding> for Embedding {
+    type Error = OpenAIError;
+
+    fn try_from(embedding: Base64Embedding) -> Result<Self, Self::Error> {
+        let bytes = general_purpose::STANDARD.decode(embedding.embedding)?;
+        let chunks = bytes.chunks_exact(4);
+        debug_assert!(chunks.remainder().len() == 0);
```

Comment:
Well this is nice.

If you're expecting this to ever fail in release build - perhaps return Err? 
 Otherwise its all good.

Reply:
I think this would be analogous to checking whether OpenAI returns an embedding that is of the correct dimensionality, or whether the response contains the correct number of embeddings. It "just won't happen". So I think it is probably safe to remove.